### PR TITLE
PHOENIX-5390 Changing anonymous inner classes in phoenix-core/main

### PR DESCRIPTION
**Pull request by nsgupta1**
@yanxinyi can you please review this PR ?

### CODE REVIEW

**Comment by swaroopak** [comment]
@nsgupta1  Could you please include the Jira number in the title like other PRs on the repo?

**Comment by yanxinyi** [comment]
can you please create a patch and submit it on the JIRA so that we can get a test run? @nsgupta1

File: phoenix-core/src/main/java/org/apache/phoenix/expression/function/InvertFunction.java [hunk 1]
```diff
@@ -141,4 +102,51 @@ public OrderPreserving preservesOrder() {
     private Expression getChildExpression() {
         return children.get(0);
     }
+
+    private static class InvertKeyPart implements  KeyPart {
```
Comment (swaroopak) [inline phoenix-core/src/main/java/org/apache/phoenix/expression/function/InvertFunction.java]: nit: extra space after implements.

File: phoenix-core/src/main/java/org/apache/phoenix/expression/function/PrefixFunction.java [hunk 2]
```diff
@@ -52,26 +50,34 @@ protected boolean extractNode() {
...
-            private final List<Expression> extractNodes = extractNode() ? Collections.<Expression>singletonList(PrefixFunction.this) : Collections.<Expression>emptyList();
+        return new PrefixKeyPart(childPart);
+    }
 
-            @Override
-            public PColumn getColumn() {
-                return childPart.getColumn();
-            }
+    private class PrefixKeyPart implements KeyPart {
+        private final List<Expression> extractNodes = extractNode() ? Collections.<Expression>singletonList(PrefixFunction.this) : Collections.<Expression>emptyList();
```
Comment (swaroopak) [inline phoenix-core/src/main/java/org/apache/phoenix/expression/function/PrefixFunction.java]: this ll give you line length warning.

**Comment by gjacoby126** [comment]
JIRA already committed, closing PR.

**Pull request closed without merging**